torch/testing/_internal/common_methods_invocations.py

```diff
@@ -7593,6 +7594,26 @@ def sample_inputs_binary_cross_entropy(op_info, device, dtype, requires_grad, lo
         for shape, kwargs in shapes_and_kwargs
     ]
 
+def sample_inputs_triplet_margin_loss(op_info, device, dtype, requires_grad, with_distance=False, **kwargs):
+    make = partial(make_tensor, (S, M), device=device, dtype=dtype, requires_grad=requires_grad)
+
+    kwargss = (
+        *[dict(margin=margin) for margin in (1e-6, 1.0, 10.0)],
+        # *[dict(p=p) for p in (0, 1, 2, 10, float("inf"))],
+        dict(swap=True),
+        *[dict(reduction=reduction) for reduction in ("mean", "sum", "none")],
+    )
+
+    sample_inputs = []
+    for kwargs in kwargss:
+        input = make()
+        args = (make(), make())
+        if with_distance:
+            kwargs["distance_function"] = nn.PairwiseDistance(p=kwargs.pop("p", 2.0), eps=kwargs.pop("eps", 1e-6))
```

Comment:
Let's make this more fun by using cosine similarity:

```
lambda x, y: 1.0 - F.cosine_similarity(x, y)
```

Reply:
I've made this change, but I'm going to revert it, since `F.cosine_similarity` only supports a subset of the dtypes of the op. See https://app.circleci.com/pipelines/github/pytorch/pytorch/419531/workflows/90e12458-df8f-4383-b995-9bcfd53157da/jobs/16818402 for some failures. Only other remedy would be to limit the supported dtypes of the `OpInfo` to the subset of the loss functional.

Reply:
Thanks @pmeier